### ISSUE TITLE
Use Distroless as base image for all Keptn services

### ISSUE THREAD

**Issue by agrimmer** [keptn/keptn#2341]
Currently, we use `alpine:3.11` as base image for most of our services. 
This distro however is "only" a community project and contains dependencies we do not need.

Therefore, this issue proposes to switch to [distroless images](https://github.com/GoogleContainerTools/distroless/tree/master/base). More precisely, to `gcr.io/distroless/base`.
With `distroless` we can decrease the size of the images.

**Definition of Done:**
- `gcr.io/distroless/base` is used as base image in all services
- Services can still be monitored with [Dynatrace](https://www.dynatrace.com/support/help/technology-support/application-software/go/)

**Comment by christian-kreuzberger-dtx** [comment]
One thing to note: golang monitoring with Dynatrace has certain requirements, which is why we use `golang-alpine` for building and `alpine` for a runtime (libc, musl, ...).

If this still works with Dynatrace, then let's reduce our image size :+1:

**Comment by christian-kreuzberger-dtx** [comment]
I've looked at the distroless containers, and I noticed the following:

1. We won't be able to have tools like `cat`, `kubectl`, `helm`, `git`, `mongodump`, ... in those containers. We either need to manually install them (which is a pain, and alpine solves this nicely) or we need to use Go-Libraries instead of the binaries which means rewriting some code. 
2. "With distroless we can decrease the size of the images." - I doubt that. I've trimmed down our docker images using alpine and removing as much of the non-essential dependencies as possible. 
For instance, take the mongodb-datastore: https://hub.docker.com/layers/keptn/mongodb-datastore/0.7.1/images/sha256-90e348deaff91c25ff2e055ab52fdaf15e260ee5745dd50b2130f7f11c0606d6?context=repo
Most of the disk space used for this image comes from `COPY` commands in the Dockerfile (e.g., the compiled binary, swagger.yaml files, config files), and not from the base image.
Another example is the cnofiguration-service, where we add `git` as a CLI tool, which obviously has some dependencies that require some disk space:
https://hub.docker.com/layers/keptn/configuration-service/0.7.1/images/sha256-3a84454b836baf813d7eba0e143f4b58ea784dbf220e26d99924a39163e55596?context=repo
3. Debugging and troubleshooting is much easier and nicer when you can just `exec` into a container and start a shell. Plus I'm not sure if distroless containers will support debugging using Skaffold and GoLand.

In summary, I don't think moving to Distroless for all our containers is the right move. Maybe it works for some of our containers, but for others we would need to do significant rewrites before we can change the base-image to distroless.

Anyway, this is just my opinion, and if you or someone else has another opinion, please share it :)

**Comment by christian-kreuzberger-dtx** [comment]
We are seeing issues with our alpine based Docker Images in certain scenarios (nslookup fails when searchdomain is misconfigured): https://github.com/keptn/keptn/issues/2951

I am seeing an immediate benefit in switching away from alpine based images to something else (distroless is an option, but let's look at other options too).

The main criteria based on my experience are:
* Deep-Monitoring with Dynatrace needs to work
* Debugging using GoLand + Skaffold needs to work
* Size of the resulting Image needs to be as small as possible (comparable to current sizes)
* Dependency Handling (e.g., installing 3rd-party packages) needs to be simple, without us having to maintain versions of those dependencies
